src/Microsoft.ML.DataView/KeyDataViewType.cs

```diff
@@ -3,55 +3,83 @@
     /// </summary>
-    public sealed class KeyType : PrimitiveDataViewType
+    public sealed class KeyDataViewType : PrimitiveDataViewType
     {
-        public KeyType(Type type, ulong count)
+        public KeyDataViewType(Type type, ulong count)
             : base(type)
         {
-            Contracts.AssertValue(type);
-            if (count == 0 || type.ToMaxInt() < count)
+            Contracts.CheckParam(IsValidDataType(type), nameof(type), "Type is not valid, it must be {0}, {1}, {2}, or {3}.", typeof(byte), typeof(ushort), typeof(uint), typeof(ulong));
+            if (count == 0 || GetMaxInt(type) < count)
                 throw Contracts.ExceptParam(nameof(count), "The cardinality of a {0} must not exceed {1}.MaxValue" +
-                    " and must be strictly positive but got {2}.", nameof(KeyType), type.Name, count);
+                    " and must be strictly positive but got {2}.", nameof(KeyDataViewType), type.Name, count);
             Count = count;
         }
 
-        public KeyType(Type type, int count)
+        public KeyDataViewType(Type type, int count)
             : this(type, (ulong)count)
         {
-            Contracts.CheckParam(0 < count, nameof(count), "The cardinality of a " + nameof(KeyType) + " must be strictly positive.");
+            Contracts.CheckParam(0 < count, nameof(count), "The cardinality of a " + nameof(KeyDataViewType) + " must be strictly positive.");
         }
 
         /// <summary>
-        /// Returns true iff the given type is valid for a <see cref="KeyType"/>. The valid ones are
+        /// Returns true iff the given type is valid for a <see cref="KeyDataViewType"/>. The valid ones are
         /// <see cref="byte"/>, <see cref="ushort"/>, <see cref="uint"/>, and <see cref="ulong"/>, that is, the unsigned integer types.
         /// </summary>
         public static bool IsValidDataType(Type type)
         {
-            Contracts.CheckValue(type, nameof(type));
             return type == typeof(byte) || type == typeof(ushort) || type == typeof(uint) || type == typeof(ulong);
         }
 
+        private static ulong GetMaxInt(Type type)
+        {
+            if (type == typeof(byte))
+                return byte.MaxValue;
+            else if (type == typeof(ushort))
+                return ushort.MaxValue;
+            else if (type == typeof(uint))
+                return uint.MaxValue;
+            else if (type == typeof(ulong))
+                return ulong.MaxValue;
+
+            return 0;
+        }
+
+        private string GetRawTypeName()
+        {
+            if (RawType == typeof(byte))
+                return NumberDataViewType.Byte.ToString();
+            else if (RawType == typeof(ushort))
+                return NumberDataViewType.UInt16.ToString();
+            else if (RawType == typeof(uint))
+                return NumberDataViewType.UInt32.ToString();
+            else if (RawType == typeof(ulong))
```

Comment:
This is one of these situations where since this specific one `uint` *is* going to be the type 99.9% of the time, we should probably structure the chained if-else statement so that this is the first condition.

Reply:
oh, right. I knew this, but forgot when I did this. Thanks for the catch.